cmd/load-runner/runner.go

```diff
@@ -211,7 +229,19 @@ func (r *LoadTestRunner) monitorMetrics(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			metrics, err := r.mimirClient.GetMetrics(ctx, r.config.Reporting.Metrics)
+			var metricsToFetch []parser.Metric
+
+			if _, ok := r.infraClient.(*DockerComposeClient); ok {
+				// For local execution, we need to fetch the specific metric format
+				metricsToFetch = []parser.Metric{
+					{Name: "rudder_load_publish_rate_per_second", Query: ""},
+				}
+			} else {
+				// For remote execution, use the configured metrics
+				metricsToFetch = r.config.Reporting.Metrics
+			}
```

Comment:
[optional] Could be simplified a bit:

```suggestion
			// For remote execution, use the configured metrics
			metricsToFetch := r.config.Reporting.Metrics
			if _, ok := r.infraClient.(*DockerComposeClient); ok {
				// For local execution, we need to fetch the specific metric format
				metricsToFetch = []parser.Metric{
					{Name: "rudder_load_publish_rate_per_second", Query: ""},
				}
			}
```